app/experimenter/experiments/api/v5/serializers.py

```diff
@@ -1030,6 +1030,19 @@ def validate(self, data):
         return data
 
     def update(self, experiment, validated_data):
+        if (
+            experiment.is_rollout
+            and not experiment.is_paused
+            and experiment.status == NimbusExperiment.Status.LIVE
+            and experiment.status_next is None
+            and experiment.publish_status == NimbusExperiment.PublishStatus.IDLE
+            and validated_data.get("publish_status")
+            != NimbusConstants.PublishStatus.REVIEW
```

Comment:
So yeah I was just thinking about this.  The `self.instance` state will be what state the experiment is in before the new changes are applied.  The states in `validated_data` are what they're going to.  From your handy dandy spreadsheet, the case we want to intercept is the case where

current state: (live, none, idle)
next state: (live, none, idle)

and then we bump idle to dirty.  I think you need to check all six here?

Reply:
Yup yup, all of them need to be there. If we don't, when we end the experiment/enrollment or reject them it'll set the state to dirty